src/xml_db/parse/entry.rs

```diff
@@ -0,0 +1,481 @@
+                        })
+                    }
+                },
+                SimpleXmlEvent::End(name) if name == "AutoType" => break,
+                _ => {
+                    return Err(XmlParseError::BadEvent {
+                        expected: "start tag or close AutoType",
+                        event: event.clone(),
+                    })
+                }
+            }
+        }
+
+        let close_tag = iterator.next().ok_or(XmlParseError::Eof)?;
+        if !matches!(close_tag, SimpleXmlEvent::End(ref tag) if tag == "AutoType") {
+            return Err(XmlParseError::BadEvent {
+                expected: "Close AutoType tag",
+                event: close_tag,
+            });
+        }
+
+        Ok(out)
+    }
+}
+
+impl FromXml for AutoTypeAssociation {
+    type Parses = Self;
+
+    fn from_xml<I: Iterator<Item = SimpleXmlEvent>>(
+        iterator: &mut Peekable<I>,
+        inner_cipher: &mut dyn Cipher,
+    ) -> Result<Self::Parses, XmlParseError> {
+        let open_tag = iterator.next().ok_or(XmlParseError::Eof)?;
+        if !matches!(open_tag, SimpleXmlEvent::Start(ref tag, _) if tag == "Association") {
+            return Err(XmlParseError::BadEvent {
+                expected: "Open Association tag",
+                event: open_tag,
+            });
+        }
+
+        let mut out = Self::default();
+
+        while let Some(event) = iterator.peek() {
+            match event {
+                SimpleXmlEvent::Start(name, _) => match &name[..] {
+                    "Window" => {
+                        out.window =
+                            SimpleTag::<Option<String>>::from_xml(iterator, inner_cipher)?.value;
+                    }
+                    "KeystrokeSequence" => {
+                        out.sequence =
+                            SimpleTag::<Option<String>>::from_xml(iterator, inner_cipher)?.value;
+                    }
+                    _ => {
+                        return Err(XmlParseError::BadEvent {
+                            expected: "valid Association child",
+                            event: event.clone(),
+                        })
+                    }
+                },
+                SimpleXmlEvent::End(name) if name == "Association" => break,
+                _ => {
+                    return Err(XmlParseError::BadEvent {
+                        expected: "start tag or close Association",
+                        event: event.clone(),
+                    })
+                }
+            }
+        }
+
+        let close_tag = iterator.next().ok_or(XmlParseError::Eof)?;
+        if !matches!(close_tag, SimpleXmlEvent::End(ref tag) if tag == "Association") {
+            return Err(XmlParseError::BadEvent {
+                expected: "Close Association tag",
+                event: close_tag,
+            });
+        }
+
+        Ok(out)
+    }
+}
+
+struct History {
+    entries: Vec<Entry>,
+}
+
+impl FromXml for History {
```

Comment:
@sseemayer is this implemented? I don't see it added to the `Entry`.
If we don't implement the entry history in this PR, I would suggest adding this to another PR, to make this one easier to review.

Reply:
With the current architecture, I essentially had to implement all the XML parsers or run into errors when parsing the XML document. The only part missing is adding fields to the appropriate `Entry`, `Group`, `Meta`, etc. fields (another breaking change), and then assigning the results of the XML parsing to those fields.